File: Códigos/Aula04.py
from clima import Clima

lista = []
nome_base = "base.csv"

try:

    leitor = open(nome_base, "r", encoding="utf-8")

    for linha in leitor:
        dados_linha = linha.split(",")
        obj_clima = Clima(dados_linha[0], dados_linha[1], dados_linha[2], dados_linha[3])

        if obj_clima not in lista:
            lista.append(obj_clima)


    for item in lista:
        print(item, end=" ")

    leitor.close()
except Exception as e:
    print("Ocorreu algum erro", e)

File: Códigos/Aula04.java
import java.util.ArrayList;

public class PrincipalClima {
    
    public static void main(String[] args) {

        ArrayList<Clima> lista = new ArrayList<Clima>();

        //simulando que estamos lendo linha a linha do arquivo
        //primeira linha
        String linha = "2020,Janeiro,Quente,muita";
        String[] dadoslinha = linha.split(",");
        Clima objClima = new Clima(dadoslinha[0], dadoslinha[1], dadoslinha[2], dadoslinha[3]);


        if(!lista.contains(objClima)){
            lista.add(objClima);
        }

        //segunda linha
        linha = "2020,Janeiro,Frio,pouca";
        dadoslinha = linha.split(",");
        objClima = new Clima(dadoslinha[0], dadoslinha[1], dadoslinha[2], dadoslinha[3]);

        
        if(!lista.contains(objClima)){
            lista.add(objClima);
        }

        for(Clima c : lista){
            System.out.println(c);
        }

    }
    
}
